redis_cache/__init__.py

```diff
@@ -204,6 +215,43 @@ def invalidate(self, *args, **kwargs):
         pipe.zrem(self.keys_key, key)
         pipe.execute()
 
+    def invalidate_partial(self, *args, **kwargs):
+        """Invalidate items in the cache based on partial matches.
+
+        A partial match is when any of the values in ``args`` are in the key's
+        ``args`` or all of the key/value pairs in ``kwargs`` are in the key's
+        ``kwargs``.
+        """
+        # First gather up a list of matching keys
+        keys = set()
+        arg_sig = signature(self.original_fn)
+        variable_args = [
+            param.name for param in arg_sig.parameters.values()
+            if param.kind is param.VAR_POSITIONAL
+        ]
+        for k in self.client.scan_iter(f'{self.get_full_prefix()}:*'):
+            kstr = k.split(":", 2)[-1]
+            key = self.deserialize_key(kstr)
+            for a in args:
+                for k1, v1 in key.items():
+                    if k1 in variable_args:
+                        if a in v1:
```

Comment:
This is kind of magical. What about strings or lists where order matters? I don't think this is safe